### ISSUE TITLE
'_at_exit' hook hangs, prevents clean shutdown

### ISSUE THREAD

**Issue by tseaver** [googleapis/gax-python#96]
After using an API instance at the Python REPL prompt, I try to exit Python using `Ctrl-D`:  the process hangs indefinitely, until I interrupt with `Ctrl-C`, at which point I get the following traceback:

```python
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/opt/Python-2.7.11/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/.tox/py27/lib/python2.7/site-packages/concurrent/futures/thread.py", line 39, in _python_exit
    t.join(sys.maxint)
  File "/opt/Python-2.7.11/lib/python2.7/threading.py", line 951, in join
    self.__block.wait(delay)
  File "/opt/Python-2.7.11/lib/python2.7/threading.py", line 359, in wait
    _sleep(delay)
KeyboardInterrupt
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/opt/Python-2.7.11/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/.tox/py27/lib/python2.7/site-packages/concurrent/futures/thread.py", line 39, in _python_exit
    t.join(sys.maxint)
  File "/opt/Python-2.7.11/lib/python2.7/threading.py", line 951, in join
    self.__block.wait(delay)
  File "/opt/Python-2.7.11/lib/python2.7/threading.py", line 359, in wait
    _sleep(delay)
KeyboardInterrupt
```

I then have to `Ctrl-C` again to get back to the shell prompt.

**Comment by tbetbetbe** [comment]
I think this is a gRPC issue being tracked as grpc/grpc#3820

**Comment by geigerj** [comment]
This appears to be fixed with gRPC release 0.15.0.

I have confirmed with a simple GAPIC example that makes a call, then attempts to exit after receiving a gRPC error response. I have also observed that, with the update, it's possible to run the gcloud-python systems test ([as described here](https://github.com/GoogleCloudPlatform/gcloud-python/issues/1869)) without indefinite hanging at the end.

**Comment by tseaver** [comment]
@geigerj Thanks!  I can confirm that the GAX-enabled pubsub system tests pass and don't hang forever.  The logging noise from gRPC is obnoxious, but not fatal.
